### DISCUSSION TITLE
Handling "Dependency Dashboard Approval" on platforms that don't support Dependency Dashboard issues

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#22298] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

I was asking what happens when a user enables "Dependency Dashboard Approval" in some way (preset, manually) on a platform that does _not_ support Dependency Dashboard issues. [^1]

According to @rarkins we essentially ruin their workflow, so we should clarify.. or improve the functionality.

I think improving the functionality is best. Silly idea: how about we raise a dummy PR somehow, which shows a suitable error message in the PR body? I assume all the platforms we support have pull requests or merge requests. After the user resolves the problem, the PR is autoclosed.

[^1]: https://github.com/renovatebot/renovate/pull/21239

**Comment by rarkins** [comment]
Two points:
1. Today, configuring approval on a non-supported platform is essentially a config error. If we chose to raise a config warning about this then we also can't make it part of config:recommended unless we think of another idea
2. At Mend we are looking to replicate the "interactive" dashboard concept in the app's companion UI, particularly for platforms like Bitbucket Cloud which can't access it today. We don't want to "block" this concept either

**Reply by HonkingGoose** [reply]
About point 1: Let's keep the "Dependency Dashboard Approval for major updates" out of the `config:recommended`/`config:best-practices` preset then. We should also raise a config warning so that users know they're using something that's not supported by their platform. Can you create an issue about raising a config error?

About point 2: I like that idea to replicate the dashboard in a companion app for platforms that don't support interactive issues. 😄

**Reply by HonkingGoose** [reply]
Another user confused themselves by _enabling_ the Dependency Dashboard, while having Issues _disabled_ on their GitHub repository:

- #23003

The current log message is at `INFO` level, maybe it should be a `WARN` or `ERROR`? We should probably also have a more descriptive and helpful log message, like this maybe:

```diff
- INFO: Cannot ensure issue because issues are disabled in this repository
+ WARN: Cannot create Dependency Dashboard issue. Make sure your Git-hosting platform supports issues, and that Issues are enabled on your repository.
```

**Reply by rarkins** [reply]
The challenge would be this scenario:
- Dashboard turned on org-wide (e.g. preset)
- Some repos have issues disabled

In this case we're warning them about something they don't want to care about. Perhaps if we enabled a setting to suppress such warnings

**Reply by HonkingGoose** [reply]
> In this case we're warning them about something they don't want to care about.

Silly comment incoming: I could argue that if the org _enables_ Dependency Dashboards in a preset, they want _all_ org repositories to use the Dashboard. 😜 So the org _wants_ to be warned about any repository not following that rule. 😄

> Perhaps if we enabled a setting to suppress such warnings

Sounds good to me. It fixes my silly comment above, because the org can control how strict they want to be. 😄

**Reply by rarkins** [reply]
So the problem is we don't differentiate between "use the dashboard if issues are enabled" or "definitely use the dashboard". I'd say that enabling issues or not is a higher/bigger decision than Renovate

**Reply by HonkingGoose** [reply]
So do you want or not want a issue for the "enabled a setting to suppress such warnings"-thing then? It's not clear from your comment what you want to do. 😉

**Reply by rarkins** [reply]
We don't need to suppress it unless we elevate it to WARN

**Reply by HonkingGoose** [reply]
I'll let you and the maintainers decide what you want to do.
